recipes-app/node-red-gpio/files/0002-Adapt-new-IOT2050-SM.patch

```diff
@@ -0,0 +1,127 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: chao zeng <chao.zeng@siemens.com>
+Date: Thu, 4 May 2023 11:51:20 +0800
+Subject: [PATCH] Adapt new IOT2050-SM
+
+Adapt new board IOT2050-SM, which reduces Arduino interfaces.
+Only keep the led and button control.
+
+Signed-off-by: chao zeng <chao.zeng@siemens.com>
+---
+ hardware/intel/mraa-gpio-ain.html  | 8 ++++++++
+ hardware/intel/mraa-gpio-din.html  | 5 +++++
+ hardware/intel/mraa-gpio-dout.html | 8 ++++++++
+ hardware/intel/mraa-gpio-led.html  | 1 +
+ hardware/intel/mraa-gpio-pwm.html  | 8 ++++++++
+ 5 files changed, 30 insertions(+)
+
+diff --git a/hardware/intel/mraa-gpio-ain.html b/hardware/intel/mraa-gpio-ain.html
+index 57b196273a52..eadf0036e236 100644
+--- a/hardware/intel/mraa-gpio-ain.html
++++ b/hardware/intel/mraa-gpio-ain.html
+@@ -15,6 +15,13 @@
+         label: function() {
+             return this.name||"A"+this.pin;
+         },
++        onpaletteadd : function() {
++            $.getJSON('mraa-gpio/'+this.id,function(data) {
++                if (data === 27) {
++                    $("#red-ui-palette-GPIO div[data-palette-type='mraa-gpio-ain']").hide();
++                }
++            });
++        },
```

Comment:
This still looks wrong as said before. I wonder why you don't catch the unsupported type in mra--gpio-ain.js, preventing the registration there in the first place.

Reply:
the .js only controlls the server side of the node.

I refactored this part to check the platform type before registering anything.